### PR TITLE
feat(core): remove deprecated `identity` module

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

Related: #3647.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by thomaseizinger** [comment]
This is currently failing because we made some errors in the previous release of `libp2p-identity 0.1.2` (unsure how that passed CI but feature flags are hard ...).

Anyway, this will pass once we upgrade to the latest `libp2p-identity` for all crates which happens in https://github.com/libp2p/rust-libp2p/pull/4037.

**Comment by thomaseizinger** [comment]
> This is currently failing because we made some errors in the previous release of `libp2p-identity 0.1.2` (unsure how that passed CI but feature flags are hard ...).
> 
> Anyway, this will pass once we upgrade to the latest `libp2p-identity` for all crates which happens in #4037.

Actually, I found a way to temporarily unblock this. We just need to activate a feature-flag that very likely will be activated anywhere somewhere in the dependency graph.

**Review by mxinden — COMMENTED** [review]
Thanks :+1:

File: core/Cargo.toml [hunk 1]
```diff
@@ -39,11 +39,9 @@ libp2p-mplex = { workspace = true }
 libp2p-noise = { workspace = true }
 multihash = { version = "0.17.0", default-features = false, features = ["arb"] }
 quickcheck = { workspace = true }
+libp2p-identity = { workspace = true, features = ["ed25519"] }
```
Comment (mxinden) [inline core/Cargo.toml:42]: Why was `libp2p-identity` added here as a dev-dependency even though it is already included as a dependency @thomaseizinger?

Comment (thomaseizinger) [reply core/Cargo.toml:42]: Originally to activate a feature but that probably became unnecessary in an iteration or something :)

Comment (thomaseizinger) [reply core/Cargo.toml:42]: See https://github.com/libp2p/rust-libp2p/pull/4040#issuecomment-1579246622.

Comment (mxinden) [reply core/Cargo.toml:42]: Thus do I understand correctly that this line is no longer needed from now on @thomaseizinger?

Comment (thomaseizinger) [reply core/Cargo.toml:42]: I think the line is needed but I think the feature can be removed!

**Pull request merged**